pkg/controller/performanceprofile/components/profile/profile.go

```diff
@@ -121,10 +147,14 @@ func validateHugepages(hugepages *v1.HugePages) error {
 		}
 	}
 
-	for _, page := range hugepages.Pages {
+	for i, page := range hugepages.Pages {
 		if page.Size != hugepagesSize1G && page.Size != hugepagesSize2M {
 			return validationError(fmt.Sprintf("the page size should be equal to %q or %q", hugepagesSize1G, hugepagesSize2M))
 		}
+
+		if err := validatePageDuplication(&page, hugepages.Pages[i+1:]); err != nil {
```

Comment:
Does it cover scenario when hp sizes goes not sequentially:

pages:
```
- count: 1
  node: 0
  size: 1G
- count: 128
  node: 0
  size: 2M
- count: 2
  node: 0
  size: 1G
```

or we don't really care about it?

Reply:
yes, it should cover all cases, the slice that I send just means that if 1 is duplicate of 3, the 3 also duplicate of 1, so no need to check again the 3 to 1 duplication.

Reply:
thanks for clarification, I need to work on my golang skills :)

Reply:
It would be nice to add this case to the unit tests

Reply:
added additional unittest